sklearn/utils/_mask.py

```diff
@@ -1,19 +1,59 @@
 import numpy as np
+from scipy import sparse
 
 from . import is_scalar_nan
 from .fixes import _object_dtype_isnan
 
 
-def _get_mask(X, value_to_mask):
-    """Compute the boolean mask X == value_to_mask."""
+def _fit_mask(X, value_to_mask):
     if is_scalar_nan(value_to_mask):
         if X.dtype.kind == "f":
-            return np.isnan(X)
+            Xt = np.isnan(X)
         elif X.dtype.kind in ("i", "u"):
             # can't have NaNs in integer array.
-            return np.zeros(X.shape, dtype=bool)
+            Xt = np.zeros(X.shape, dtype=bool)
         else:
             # np.isnan does not work on object dtypes.
-            return _object_dtype_isnan(X)
+            Xt = _object_dtype_isnan(X)
     else:
-        return X == value_to_mask
+        Xt = X == value_to_mask
+
+    return Xt
+
+
+def _get_mask(X, value_to_mask, reconstruct_sparse=False):
+    """Compute the boolean mask X == value_to_mask.
+
+    Parameters
+    ----------
+    X : {array-like, sparse matrix}, shape (n_samples, n_features)
+        Input data, where ``n_samples`` is the number of samples and
+        ``n_features`` is the number of features.
+
+    value_to_mask : {int, float, nan}
+                    The values for which should be maked in X.
+
+    reconstruct_sparse : bool, default=False
+                         Whether or not we need to reconstruct sparse matrix.
+                         If True, X is considered considered sparse and sparse
```

Comment:
indentation here as well

Reply:
instead of `reconstruct_sparse` I think we should call it `sparse`. It will indicate if whether or not we return a sparse matrix of the same format as `X`

Reply:
We the problem with calling it `sparse` is that we are importing the module named `sparse` hence it would generate error. Should I name it `Sparse` instead?

Reply:
we can import with alias `from scipy import sparse as sp`